### PR TITLE
Support HitCircleOverlayAboveNumer typo for old legacy skins

### PR DESCRIPTION
**Pull request by LittleEndu**

### CODE REVIEW

**Comment by peppy** [comment]
can you do this i the transformer instead? and comment a link referencing where this is documented (on the wiki or otherwise)

**Comment by mcendu** [comment]
[Per wiki:](https://osu.ppy.sh/help/wiki/Skinning/skin.ini#%5Bgeneral%5D)

> Old command: `HitCircleOverlayAboveNumer` (with typo) still works for legacy support

**Comment by peppy** [comment]
Yep, that link should be inline in the code as a reference!

**Comment by LittleEndu** [comment]
> can you do this in the transformer instead?

You mean replace the typo when the skin is first imported?

**Comment by mcendu** [comment]
> 
> 
> > can you do this in the transformer instead?
> 
> You mean replace the typo when the skin is first imported?

imo in ```ManiaLegacySkinTransformer.GetConfig`2``` when the entry without typo isn't found, we should try to get the entry with typo instead of returning null. This can make code that needs to check this property look cleaner.

File: osu.Game.Rulesets.Osu/Skinning/OsuLegacySkinTransformer.cs [hunk 1]
```diff
@@ -132,6 +132,12 @@ public Drawable GetDrawableComponent(ISkinComponent component)
                                 return SkinUtils.As<TValue>(new BindableFloat(LEGACY_CIRCLE_RADIUS));
 
                             break;
+
+                        case OsuSkinConfiguration.HitCircleOverlayAboveNumber:
+                            // Quote from https://osu.ppy.sh/help/wiki/Skinning/skin.ini#%5Bgeneral%5D
+                            // Old command: HitCircleOverlayAboveNumer (with typo) still works for legacy support
+                            var rv = source.GetConfig<OsuSkinConfiguration, TValue>(OsuSkinConfiguration.HitCircleOverlayAboveNumber);
```
Comment (mcendu) [inline osu.Game.Rulesets.Osu/Skinning/OsuLegacySkinTransformer.cs:139]: Could be inlined.

File: osu.Game.Rulesets.Osu/Skinning/OsuLegacySkinTransformer.cs [hunk 2]
```diff
@@ -132,6 +132,12 @@ public Drawable GetDrawableComponent(ISkinComponent component)
...
+                            return rv ?? source.GetConfig<OsuSkinConfiguration, TValue>(OsuSkinConfiguration.HitCircleOverlayAboveNumer);
```
Comment (LittleEndu) [inline osu.Game.Rulesets.Osu/Skinning/OsuLegacySkinTransformer.cs:140]:
```suggestion
                            return source.GetConfig<OsuSkinConfiguration, TValue>(OsuSkinConfiguration.HitCircleOverlayAboveNumber) ?? source.GetConfig<OsuSkinConfiguration, TValue>(OsuSkinConfiguration.HitCircleOverlayAboveNumer);
                            
```

@mcendu You mean like this?

Comment (bdach) [reply osu.Game.Rulesets.Osu/Skinning/OsuLegacySkinTransformer.cs:140]: I'd say that's better (also gets rid of the very cryptic `rv` variable name).

File: osu.Game.Rulesets.Osu/Skinning/OsuLegacySkinTransformer.cs [hunk 3]
```diff
@@ -136,7 +136,7 @@ public Drawable GetDrawableComponent(ISkinComponent component)
                         case OsuSkinConfiguration.HitCircleOverlayAboveNumber:
                             // Quote from https://osu.ppy.sh/help/wiki/Skinning/skin.ini#%5Bgeneral%5D
                             // Old command: HitCircleOverlayAboveNumer (with typo) still works for legacy support
-                            return source.GetConfig<OsuSkinConfiguration, TValue>(OsuSkinConfiguration.HitCircleOverlayAboveNumber) ?? source.GetConfig<OsuSkinConfiguration, TValue>(OsuSkinConfiguration.HitCircleOverlayAboveNumer);                       
+                            return source.GetConfig<OsuSkinConfiguration, TValue>(OsuSkinConfiguration.HitCircleOverlayAboveNumber) ?? source.GetConfig<OsuSkinConfiguration, TValue>(OsuSkinConfiguration.HitCircleOverlayAboveNumer);
```
Comment (mcendu) [inline osu.Game.Rulesets.Osu/Skinning/OsuLegacySkinTransformer.cs:139]: That’s a _long_ line. Better put a new line before the `??`.

**Review by peppy — APPROVED** [review]

File: osu.Game.Rulesets.Osu/Skinning/OsuLegacySkinTransformer.cs [hunk 4]
```diff
@@ -132,6 +132,12 @@ public Drawable GetDrawableComponent(ISkinComponent component)
                                 return SkinUtils.As<TValue>(new BindableFloat(LEGACY_CIRCLE_RADIUS));
 
                             break;
+
+                        case OsuSkinConfiguration.HitCircleOverlayAboveNumber:
+                            // See https://osu.ppy.sh/help/wiki/Skinning/skin.ini#%5Bgeneral%5D
+                            // HitCircleOverlayAboveNumer (with typo) should still be supported for now.
+                            return source.GetConfig<OsuSkinConfiguration, TValue>(OsuSkinConfiguration.HitCircleOverlayAboveNumber) ??
+                                   source.GetConfig<OsuSkinConfiguration, TValue>(OsuSkinConfiguration.HitCircleOverlayAboveNumer);
```
Comment (smoogipoo) [inline osu.Game.Rulesets.Osu/Skinning/OsuLegacySkinTransformer.cs:140]: Not a big deal but in cases like this I would rather send down a string in the first place, i.e. `source.GetConfig<string, TValue>("HitCircleOverlayAboveNumer");`, as it's a hack that shouldn't exist in the config enum.

**Pull request merged**